plutus-core/plutus-core/src/PlutusCore/Check/Scoping.hs

```diff
@@ -279,14 +301,14 @@ data ScopeError
     | NameChangedItsScope !ScopedName !ScopedName
     | NameUnexpectedlyDisappeared !ScopedName !ScopedName
     | NameUnexpectedlyStayed !ScopedName
-    | DuplicateBindersInTheInput !(Set ScopedName)
-    | DuplicateBindersInTheOutput !(Set ScopedName)
-    | OldBindingsDiscordWithBoundVariables !(Set ScopedName)
-    | OldBindingsDiscordWithOutOfScopeVariables !(Set ScopedName)
-    | NewBindingsDiscordWithBoundVariables !(Set ScopedName)
-    | OldBindingsClashWithFreeVariables !(Set ScopedName)
-    | OldBindingsClashWithNewBindings !(Set ScopedName)
-    | NewBindingsClashWithFreeVariabes !(Set ScopedName)
+    | DuplicateBindersInTheInput !(Set ScopedName) !(Set ScopedName)
+    | DuplicateBindersInTheOutput !(Set ScopedName) !(Set ScopedName)
+    | DisappearedBindingsDiscordWithBoundVariables !(Set ScopedName) !(Set ScopedName)
+    | DisappearedBindingsDiscordWithOutOfScopeVariables !(Set ScopedName) !(Set ScopedName)
```

Comment:
Makes it much easier to debug stuff when you can see what exactly disappeared and what the out of scope variables are. Same for all the other errors.